samples/bluetooth/peripheral_cts_client/src/main.c

```diff
@@ -0,0 +1,356 @@
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	bt_conn_auth_pairing_confirm(conn);
+
+	printk("Pairing confirmed: %s\n", addr);
+}
+
+static void pairing_complete(struct bt_conn *conn, bool bonded)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	printk("Pairing completed: %s, bonded: %d\n", addr, bonded);
+}
+
+static void pairing_failed(struct bt_conn *conn, enum bt_security_err reason)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+
+	bt_addr_le_to_str(bt_conn_get_dst(conn), addr, sizeof(addr));
+
+	printk("Pairing failed conn: %s, reason %d\n", addr, reason);
+}
+
+static struct bt_conn_auth_cb conn_auth_callbacks = {
+	.cancel = auth_cancel,
+	.pairing_confirm = pairing_confirm,
+	.pairing_complete = pairing_complete,
+	.pairing_failed = pairing_failed
+};
+
+/**@brief Function for handling the Current Time Service errors.
+ *
+ * @param[in] p_evt  Event received from the Current Time Service client.
+ */
+static void current_time_print(bt_gatt_cts_c_evt_t *p_evt)
+{
+	printk("\nCurrent Time:\n");
+	printk("\nDate:\n");
+
+	printk("\tDay of week   %s\n",
+	       day_of_week[p_evt->current_time.exact_time_256.day_of_week]);
+
+	if (p_evt->current_time.exact_time_256.day == 0) {
+		printk("\tDay of month  Unknown\n");
+	} else {
+		printk("\tDay of month  %u\n",
+		       p_evt->current_time.exact_time_256.day);
+	}
+
+	printk("\tMonth of year %s\n",
+	       month_of_year[p_evt->current_time.exact_time_256.month]);
+	if (p_evt->current_time.exact_time_256.year == 0) {
+		printk("\tYear          Unknown\n");
+	} else {
+		printk("\tYear          %u\n",
+		       p_evt->current_time.exact_time_256.year);
+	}
+	printk("\nTime:\n");
+	printk("\tHours     %u\n", p_evt->current_time.exact_time_256.hours);
+	printk("\tMinutes   %u\n", p_evt->current_time.exact_time_256.minutes);
+	printk("\tSeconds   %u\n", p_evt->current_time.exact_time_256.seconds);
+	printk("\tFractions %u/256 of a second\n",
+	       p_evt->current_time.exact_time_256.fractions256);
+
+	printk("\nAdjust reason:\n");
+	printk("\tDaylight savings %x\n",
+	       p_evt->current_time.adjust_reason
+		       .change_of_daylight_savings_time);
+	printk("\tTime zone        %x\n",
+	       p_evt->current_time.adjust_reason.change_of_time_zone);
+	printk("\tExternal update  %x\n",
+	       p_evt->current_time.adjust_reason.external_reference_time_update);
+	printk("\tManual update    %x\n",
+	       p_evt->current_time.adjust_reason.manual_time_update);
+}
+
+static void bt_cts_c_evt_handler(struct bt_gatt_cts_c *cts_c,
+				 bt_gatt_cts_c_evt_t *p_evt)
+{
+	switch (p_evt->evt_type) {
+	case BT_GATT_CTS_C_EVT_CURRENT_TIME:
+		printk("Current Time received.\n");
+		current_time_print(p_evt);
+		break;
+
+	case BT_GATT_CTS_C_EVT_INVALID_TIME:
+		printk("Invalid Time received.\n");
+		break;
+
+	default:
+		/* No implementation needed. */
+		break;
+	}
+}
+
+static int cts_c_init(void)
```

Comment:
If you pass event handler as function parameter instead of structure this additional function won't be needed

Reply:
Done.